src/fastmap.cpp

```
@@ -381,15 +412,58 @@ ktp_data_t *kt_pipeline(void *shared, int step, void *data, mem_opt_t *opt, work
+            }
+
+            if (aux->opt->meth_mode && g_meth_bam_writer != NULL) {
+                /* Gather all bam1_t* in the QNAME group, propagate QC fail, emit. */
+                int total = 0;
+                for (int k = 0; k < group_size; ++k) total += ret->seqs[i+k].n_bams;
+                if (total > 0) {
+                    struct bam1_t **group = (struct bam1_t **)malloc(total * sizeof(struct bam1_t *));
+                    int idx = 0;
+                    for (int k = 0; k < group_size; ++k) {
+                        for (int j = 0; j < ret->seqs[i+k].n_bams; ++j) {
+                            group[idx++] = (struct bam1_t *)ret->seqs[i+k].bams[j];
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)